test/parallel_exec.jl

```diff
@@ -663,28 +663,14 @@ end
 
 # pmap tests. Needs at least 4 processors dedicated to the below tests. Which we currently have
 # since the parallel tests are now spawned as a separate set.
-function unmangle_exception(e)
-    while any(x->isa(e, x), [CompositeException, RemoteException, CapturedException])
-        if isa(e, CompositeException)
-            e = e.exceptions[1].ex
-        end
-        if isa(e, RemoteException)
-            e = e.captured.ex
-        end
-        if isa(e, CapturedException)
-            e = e.ex
-        end
-    end
-    return e
-end
 
 # Test all combinations of pmap keyword args.
 pmap_args = [
                 (:distributed, [:default, false]),
                 (:batch_size, [:default,2]),
-                (:on_error, [:default, e -> unmangle_exception(e).msg == "foobar"]),
-                (:retry_delays, [:default, fill(0.01, 1000)]),
-                (:retry_check, [:default, (s,e) -> (s,unmangle_exception(e).msg == "foobar")]),
+                (:on_error, [:default, e -> (e.msg == "foobar" ? true : rethrow(e))]),
+                (:retry_delays, [:default, fill(0.001, 1000)]),
+                (:retry_check, [:default, (s,e) -> (s,endswith(e.msg,"foobar"))]),
```

Comment:
Helps differentiate between `on_error` and `retry_check` callbacks. One of the test functions below throws "notfoobar".